src/devices/cpu/i386/i386.cpp

```diff
@@ -3944,9 +3959,76 @@ void i386_device::device_reset()
 
 	m_CPL = 0;
 
+	m_auto_clear_RF = true;
+
 	CHANGE_PC(m_eip);
 }
 
+inline void i386_device::dri_changed()
+{
+	int dr;
+	for(dr = 0; dr < 4; dr++)
+	{
+		int dr_enabled = (m_dr[7] & (1 << (dr << 1))) || (m_dr[7] & (1 << ((dr << 1) + 1))); // Check both local enable AND global enable bits for this breakpoint.
+		if(dr_enabled)
+		{
+			if(m_dr_breakpoints[dr]) m_dr_breakpoints[dr]->remove();
```

Comment:
Shouldn't you remove the previous breakpoint even if dr_enabled == 0 ?  Or how is a breakpoint supposed to be removed?

Reply:
Oof, good point.